submit_and_compare/submit_and_compare.py

```diff
@@ -300,6 +303,16 @@ def student_submit(self, submissions, suffix=''):
                 'submit_class': self._get_submit_class(),
                 'used_attempts_feedback': self._get_used_attempts_feedback(),
             }
+	elif self.is_past_due():
+	    LOG.error(
+                'This problem is past due',
+            )
+            result = {
+                'success': False,
+                'problem_progress': self._get_problem_progress(),
+                'submit_class': self._get_submit_class(),
+                'used_attempts_feedback': self._get_used_attempts_feedback(),
+            }
```

Comment:
now that we're doubling a lot of this code, let's see if we can't coalesce it and make it more elegant.